### PR TITLE
Move "allows remote indices" to a request property

### PR DESCRIPTION
**Pull request by albertzaharovits**
Some requests can reference index names on a remote cluster, yet most don't.
When (locally) authorizing requests that CAN NOT reference remote indices, the security code takes a (cached) shortcut to first check if the request is authorized for any index, before doing the in-detail authorization on the actual indices.

This PR moves the "allows remote indices" property to the indices request away from Security code, with the goal that it's useful to non-security code as well (rather than duplicating it in the other places).

NB It's rare that non-Security code filters requests of different types by this property, if that exists at all. But conceptually I don't think the Security code should decide which requests reference remote indices.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/IndicesRequest.java [hunk 1]
```diff
@@ -29,6 +29,12 @@
      */
     IndicesOptions indicesOptions();
 
+    /** Determines whether the request can contain indices on a remote cluster.
+     */
```
Comment (ywangd) [inline server/src/main/java/org/elasticsearch/action/IndicesRequest.java:33]: Nit
```suggestion
    /** 
     * Determines whether the request can contain indices on a remote cluster.
     */
```

**Review by ywangd — APPROVED** [review]
LGTM

There might be a historical reason why this logic was in security code. I don't know whether the reason still exists. But the change looks right to me. The fact that we had to use the class name of the Eql request is a pretty strong suggestion for this change.

**Comment by albertzaharovits** [comment]
Thank you for the quick review!

> There might be a historical reason why this logic was in security code. I don't know whether the reason still exists. But the change looks right to me. The fact that we had to use the class name of the Eql request is a pretty strong suggestion for this change.

I think that in practice only Security code has to deal with requests of multiple types and differentiate them by this criterion.
But in theory it is a concern that code outside Security could be interest in. Indeed it hasn't bothered me enough until using hardcoded class names.

**Comment by albertzaharovits** [comment]
@elasticmachine update branch please

**Comment by albertzaharovits** [comment]
@elasticmachine update branch please

**Comment by albertzaharovits** [comment]
@elasticmachine update branch please

**Comment by albertzaharovits** [comment]
FIPS builds fail for unrelated reason https://github.com/elastic/elasticsearch/pull/74692

**Pull request merged**